lib/ImportFromC/HeaderToModel.cpp

```diff
@@ -157,41 +158,61 @@ DeclVisitor::DeclVisitor(TupleTree<model::Binary> &Model,
   AnalysisOption(AnalysisOption) {
 }
 
-// Parse ABI from the annotate attribute content.
-static std::optional<std::string> getABI(llvm::StringRef ABIAnnotate) {
-  if (not ABIAnnotate.startswith(ABIAnnotation))
-    return std::nullopt;
+template<typename Type>
+static std::optional<llvm::StringRef>
+parseStringAnnotation(const Type &Declaration, llvm::StringRef Prefix) {
+  std::optional<llvm::StringRef> Result;
+  if (Declaration.template hasAttr<clang::AnnotateAttr>()) {
+    for (auto &Attribute : Declaration.getAttrs()) {
+      if (auto *Cast = llvm::dyn_cast<clang::AnnotateAttr>(Attribute)) {
+        llvm::StringRef Annotation = Cast->getAnnotation();
+        if (not Annotation.startswith(Prefix))
+          continue;
+
+        llvm::StringRef Value = Annotation.substr(Prefix.size());
+        if (Result.has_value() && Result.value() != Value) {
+          std::string Error = "Multiple conflicting annotation values found: '"
+                              + Result.value().str() + "' and '" + Value.str()
+                              + "'";
+          revng_log(Log, Error.c_str());
+          return std::nullopt;
+        }
 
-  return std::string(ABIAnnotate.substr(ABIAnnotation.size()));
-}
+        Result = Value;
+      }
+    }
+  }
 
-static model::Architecture::Values getRawABIArchitecture(llvm::StringRef ABI) {
-  revng_assert(ABI.starts_with(RawABIPrefix));
-  return model::Architecture::fromName(ABI.substr(RawABIPrefix.size()));
+  return Result;
 }
 
-// Parse location of parameter (`reg:` or `stack`).
-static std::optional<std::string> getLoc(llvm::StringRef Annotation) {
-  if (Annotation == "stack")
-    return Annotation.str();
+template<typename Type>
+static std::optional<uint64_t>
+parseIntegerAnnotation(const Type &Declaration, llvm::StringRef Prefix) {
+  std::optional<llvm::StringRef> Result = parseStringAnnotation(&Declaration,
```

Comment:
I think this doesn't compile. Have you checked with `orc check branch`?

Reply:
Fuck, I squashed the fix into the wrong commit (`HeaderToModel: add padding annotation support`). Thanks for noticing.

Reply:
Fixed!